src/Raven.Client/Documents/Session/AbstractDocumentQuery.Timings.cs

```diff
@@ -6,15 +6,22 @@ public abstract partial class AbstractDocumentQuery<T, TSelf>
     {
         protected QueryTimings QueryTimings;
 
+        protected FlagHolder ShouldIncludeTimings = new FlagHolder { Value = false };
```

Comment:
Why is this needed?

Reply:
This property has to be nonprimitive (with reference), because if you update the bool in 1 DocumentQuery you want it'll be updated on all of its clones.
'bool' is primitive and changing it in the original DocumentQuery won't cause changing it in the clones.

Reply:
@shaharhikri, let's put the `bool` inside `QueryTimings` and than you don't need the `FlagHolder`